chia/wallet/puzzles/genesis-by-coin-id-with-0.clvm

```diff
@@ -1,39 +1,27 @@
 ; This is a "genesis checker" for use with cc.clvm.
 ;
-; This checker allows new ccs to be created if they have a particular
-; coin id as parent; or created by anyone if their value is 0.
+; This checker allows new CATs to be created if they have a particular coin id as parent
+;
+; The genesis_id is curried in, making this lineage_check program unique and giving the CAT it's uniqueness
 
 (mod (
-      genesis-id
-      lineage-proof-parameters
-      my-coin-info
-      (parent-coin zero-parent-inner-puzzle-hash)
+      GENESIS_ID
+      Truths ; These should be the same CAT truths, but minus the opaque solution which gets passed below
+      delta
+      lineage_check_result
+      inner_conditions
+      _
     )
 
-    ;; boolean or macro
-    ;; This lets you write something like (if (or COND1 COND2 COND3) (do-something) (do-something-else))
-    (defmacro or ARGS
-        (if ARGS
-            (qq (if (unquote (f ARGS))
-                1
-                (unquote (c or (r ARGS)))
-            ))
-        0)
-    )
+    (include cat_truths.clib)
 
-    (defun-inline main (
-      genesis-id
-      my-coin-info
+    ; Returns nil since we don't need to add any conditions
```

Comment:
This is not at all what seems to be happening. Can you clarify?

Reply:
Yeah that's an outdated comment, we'll fix it :+1: